Wip nctalkproxyd

rzerres opened the pull request:
Hey Felix,

this PR is based on your  **nextcloud-talk-bot-framework** related work.

#  Code review
I did review the code an like to reccommend the following changes. Have a look at commit
**a3e8716** .  All hunks try to streamline naming of constants and variables to be more in line with the nextcloud relations. This is ment as RFC. So, id like to shorten the prefix part of the _"Nextcloud-Talk"_ related images and use a common prefix (like the choosen nctalk-<image>

# Archlinux 
The archlinux specific changes will follow from commits b6190e2  up to 6591a7a.
The `makepkg`  helper-script has a hardcoded path to $pkgdir which resolves to `pkg`. Since your go modules are bundled inside `pkg` i need to copy a build-dir, compile and package. In this run,
the `git clone` created pkg-dir will be cleaned. The correct source is under ./src/nctalkproxyd-git/. 
Until the commits are merged into origin, i have a commit, that patches the PKGBUILD script to pull from my fork wip-nctalkproxyd branch ( 2d9a3c7 ). This enables a clean run, but shouldn't be merged. 

# OCI
In line with the OCI adaptions to the old `nextcloud-talk-jitsi-bot` you will find commits updates for the new framework stuff (OCI.md, CONTRIBUTION.md ). I did upload the `nctalkproxy` image to Quay.io

If i misinterpreded the usage of ENV, parameters and config-file settings please give me a light.
Once the image is build, either docker or OCI-format, the config-file is pretty hard to use and change. The defaults have to be overwitten via commandline parameter, or ENV variables. I have adapted the names. Hope this makes sense to you.

cheers Ralf

PS: work on **nextcloud-talk-bot-jitsi** is underway and should present an image ''nctalk-bot".

rzerres committed:
nctalkproxyd: rename module and constants

* make names consistent to comply with nextcloud API
  (Nextcloud is always aliased as 'nc')
* rename constants to be more intuitive
* adapt configFileDefault (nctalkproxyd.yaml)

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

Commit:
docs: Add CoC

Commit:
fix: Use correct links to Docker image

Commit:
docs: Add video introduction

rzerres committed:
build_container.sh: scriting the image build process

* create on OCI 'nctalkproxyd' image
* cleanup build packages after compile run
* option: allow manual intervention before committing the image
* option: upload image to repo provider
* handle rootless and root mode respectively

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
PKGBUILD.git: create new package 'nctalkproxyd'

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
OCI.md: Update documentation

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
.gitignore: update packaging artefacts

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
systemd.unit: introduce new service unit for nctalkproxyd binary

* create service unit where its name is in line with the package name
* reflact usage of a config file
* as an alternative make usage of environment variables

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
PKGBUILD: temporarily use fork repository until merged

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
PKGBUILD.git: solve handling of pkgdir

* source does use a 'pkg' directory (go packages)
  packaging process needs to build in a distictable 'pkgdir'
* hint message advertising packaging directory

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
main.go: logging config file / default config file

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
CONTRIBUTING.md: update documentation to meet new image

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

pojntfx commented on README.md:
```diff
@@ -41,17 +45,17 @@ To then start using your bot, you can connect your bot to `nxtalkproxyd` like so
     -e BOT_JITSI_BOT_NAME=botusername \
     -e BOT_JITSI_SLEEP_TIME=20 \
-    -e BOT_NXTALKPROXYD_ADDR=nxtalkproxy:1969 \
+    -e BOT_NXTALKPROXYD_ADDR=nxtalkproxyd:1969 \
```
Once these env variables have changed their names, they should probably also be changed over in the actual bot repo.

rzerres replied:
Which is fine for me ...

> This should be `https://pojntfx.github.io/nextcloud-talk-bot-framework/` - simply because the repo is on this path and GitHub doesn't let you configure it.

rzerres replied:
Ok. Will patch it.

rzerres committed:
main.go: refine variable name "addrLocale" -> "addrLocal"

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
main.go: update reference to github repo

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
OCI.md: update variable names

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
README.md: update documentation text

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres commented:
I'd like discuss some issues related to `nextcloud-talk-bot-jitsi` | `jitsi-meet-node-client` | `nctalkproxyd` in a meet.jit.si session. When you are available, may you open an invitation?

pojntfx commented:
Sure! I'll keep https://meet.jit.si/nctalkproxyd open in the background, feel free to join right now if you want to ;)

rzerres commented:
We did discuss the missing API to just read in the last messages. Probably you did see
https://nextcloud-talk.readthedocs.io/en/latest/chat/ (Receive chat messages of a conversation)
If not, may the field 'lastKnownMessageID' help in combination with 'includeLastKnown' and 'lookIntoFuture' ?

rzerres committed:
OCI.md: typo correction

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
main.go: improve log messages

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
main: improve handling of commandline args

* update structure to be conformant with corba upstream
* insert logging in debug level
* insert logging in info level
* handle path search for config file
* handle environment variables, always taking precedence
* if default config file is missing, proceed with buildin settings

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
go.sum: update checksum file refering to new package versions

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
build_container.sh: update reflecting upstream changes

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
README.md: introduce images to illustrate the framework structure

* extend the documentation text
* upload nctalkbot-framework.svg
* upload nctalkbot-framework.png
* use png, since the svg isn't rendered correctly
* naming correction

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
client: reorder functions, add comments

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
client: enhance structure Room

* reflact new API: /ocs/v2.php/apps/spreed/api/v2
  https://nextcloud-talk.readthedocs.io/en/latest/conversation/#get-user-s-conversations
* incorporate new sensitive fiedls to the structure
* may allow to improve efficency in given loop
  to scan new rooms/chats
* base to differenciate the conversation type
  https://nextcloud-talk.readthedocs.io/en/latest/constants/#Conversation-types

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
client: enhance structure Chat

* incorporate new sensitive fiedls to the structure
* may allow to improve efficency in given loop
  to interact with chats
* may improve rendering with new messageParameters
  which are handled as RichObjectStrings

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
images: enhance images to include jitsi-meet-node-client

* new image: jitsi_meet_mode_client.png
  created browser windows via JitsiMeetExternalAPI
* new jitsi-meet-node
* new methods for client jitsi

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres commented:
Hey Felix,

if you find the time, please give me a short feedback. We might go for another Jitsi-Meeting?
Quite some work has gone by. Probably plenty of stuff to discuss?
If this channel isn't appropriate, i can send you an email.

Ralf

pojntfx commented on cmd/constants.go:
```diff
@@ -1,8 +1,8 @@
 const (
-	NXTalkProxyDDefaultLaddr = ":1969"                      // NXTalkProxyDDefaultLaddr is the default Host:port of `nxtalkproxyd`.
-	ConfigurationFileDocs    = "Configuration file to use." // ConfigurationFileDocs is the documentation for the configuration file flag.
+	NcTalkProxydDefaultAddrLocal = "localhost:1969"                       // NcTalkProxydDefaultAddrLocal is the default Host:port of `nctalkproxyd`.
```
This should really be `:1969`; otherwise the proxy is not reachable from outside a CRI like Docker.

rzerres replied:
is integrated

pojntfx reviewed (changes requested):
Thanks so much your PR! Really appreciate the graphics; these should make adoption for new users much easier. Just one small nitpick, see the comment.

Commit:
Merge branch 'master' into wip-nctalkproxyd

rzerres committed:
constants.go: port update to work in cluster environment

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

rzerres committed:
Merge branch 'wip-nctalkproxyd' of https://github.com/rzerres/nextcloud-talk-bot-framework into wip-nctalkproxyd
